crates/ruma-events/src/room/message.rs

```diff
@@ -152,6 +155,133 @@ impl RoomMessageEventContent {
         forward_thread: ForwardThread,
         add_mentions: AddMentions,
     ) -> Self {
+        self = self.make_reply_fallback(original_message.into());
+
+        let original_thread_id = as_variant!(
+            &original_message.content.relates_to,
+            Some(Relation::Thread(thread)) => thread
+        )
+        .filter(|_| forward_thread == ForwardThread::Yes)
+        .map(|thread| thread.event_id.clone());
+
+        let original_user_mentions = if add_mentions == AddMentions::Yes {
+            Some(
+                original_message
+                    .content
+                    .mentions
+                    .as_ref()
+                    .map(|m| m.user_ids.clone())
+                    .unwrap_or_default(),
+            )
+        } else {
+            None
+        };
```

Comment:
This one on the other hand can be a use of `bool::then`:

```suggestion
        let original_user_mentions = (add_mentions == AddMentions::Yes).then(|| {
            original_message
                .content
                .mentions
                .as_ref()
                .map(|m| m.user_ids.clone())
                .unwrap_or_default(),
        });
```